sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -1636,26 +1636,33 @@ class Analyzer(
                     s"its class is ${other.getClass.getCanonicalName}, which is not a generator.")
               }
             }
-          case u @ UnresolvedFunction(funcId, children, isDistinct) =>
+          case u @ UnresolvedFunction(funcId, arguments, isDistinct, filter) =>
             withPosition(u) {
-              v1SessionCatalog.lookupFunction(funcId, children) match {
+              v1SessionCatalog.lookupFunction(funcId, arguments) match {
                 // AggregateWindowFunctions are AggregateFunctions that can only be evaluated within
                 // the context of a Window clause. They do not need to be wrapped in an
                 // AggregateExpression.
                 case wf: AggregateWindowFunction =>
                   if (isDistinct) {
                     failAnalysis(
                       s"DISTINCT specified, but ${wf.prettyName} is not an aggregate function")
+                  } else if (filter.isDefined) {
```

Comment:
If both words specified, only the `DISTINCT` is shown in error messages, so how about this?
```
                  val notSupportedWords = (if (isDistinct) "DISTINCT" :: Nil else Nil) ++
                    (if (filter.isDefined) "FILTER" :: Nil else Nil)
                  if (notSupportedWords.nonEmpty) {
                    failAnalysis(
                      s"${notSupportedWords.mkString(" and ")} specified, but ${wf.prettyName} " +
                        "is not an aggregate function")
                  } else {
                    wf
                  }
```

Reply:
OK.